p2p/host/basic/basic_host.go

```diff
@@ -121,6 +135,15 @@ func (h *BasicHost) newStreamHandler(s inet.Stream) {
 		s.Close()
 		return
 	}
+
+	if h.NegotiateTimeout != 0 {
+		if err := s.SetDeadline(time.Time{}); err != nil {
```

Comment:
Yeap, documentation for SetDealine states:
`A zero value for t means I/O operations will not time out.`